### ISSUE TITLE
Forked-daapd fails to start randomly.

### ISSUE THREAD

**Issue by Ulrichgau** [ejurgensen/forked-daapd#1095]
Randomly in about 20% of the starts forked-daapd doesn't start at all. After shutdown the raspi and restart forked-daapd starts in 20% of the cases. In the log there is 
[Fatal]   httpd: Could not bind to port 3689 (forked-daapd already running?)

- version of forked daapd: 27.2 (also with 27.1)
- platform raspberrypi (buster)

Please try to provide the following:
Any suggestions how to get more information?

**Comment by ejurgensen** [comment]
> Randomly in about 20% of the starts forked-daapd doesn't start at all

How precisely are you starting forked-daapd in this case?

**Comment by Ulrichgau** [comment]
I switch on my raspi and wait 2 mins. Then I start my mpd client or the webclient. If there is no connection I restart my raspi. In seldom cases I have to restart twice.--Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 18.09.20, 15:19 schrieb ejurgensen <notifications@github.com>:

   
   
   Randomly in about 20% of the starts forked-daapd doesn't start at all 
   
  How precisely are you starting forked-daapd in this case?

**Comment by Ulrichgau** [comment]
Maybe I misunderstood your question. The start of forked-daapd is made via an init.d script which has been installed automatically if I remember correctly.--Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 18.09.20, 15:19 schrieb ejurgensen <notifications@github.com>:

   
   
   Randomly in about 20% of the starts forked-daapd doesn't start at all 
   
  How precisely are you starting forked-daapd in this case?

**Comment by ejurgensen** [comment]
> Randomly in about 20% of the starts forked-daapd doesn't start at all. After shutdown the raspi and restart forked-daapd starts in 20% of the cases.

I'm having trouble understanding the math here. In the first sentence you say that it starts in 80% of the cases, but in the second in 20%. So I though you were starting in different ways. But from the above it sounds like you are always just rebooting?

**Comment by ejurgensen** [comment]
I have tried rebooting my Buster a bunch of times, but I could not reproduce. The most obvious reason for your issue is of course that something else is using port 3689, as the log message says. I'm not sure what the best way to determine that is.

**Comment by Ulrichgau** [comment]
I found a solution. With StartLimitBurst=5 instead of 2 in forked-daapd.service it's working well.--Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 18.09.20, 19:35 schrieb ejurgensen <notifications@github.com>:

   
  I have tried rebooting my Buster a bunch of times, but I could not reproduce. The most obvious reason for your issue is of course that something else is using port 3689, as the log message says. I'm not sure what the best way to determine that is.
